diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -396,22 +396,18 @@ controller_interface::CallbackReturn DiffDriveController::on_configure(
   std::string tf_prefix = "";
   if (params_.tf_frame_prefix_enable)
   {
-    if (params_.tf_frame_prefix != "")
+    if (!params_.tf_frame_prefix.empty())
     {
       tf_prefix = params_.tf_frame_prefix;
     }
     else
     {
       tf_prefix = std::string(get_node()->get_namespace());
-    }
-
-    if (tf_prefix == "/")
-    {
-      tf_prefix = "";
-    }
-    else
-    {
-      tf_prefix = tf_prefix + "/";
+      tf_prefix.erase(0, 1);
```

Comment:
Why do you always want to erase the first character? If the prefix is parsed without '/' then it removes the wrong character right?

I see that you did that when you are using node namespace, however, I think you need to correct the users tf_prefix as well, if he adds a '/' as well. Try to do same logic as done in #1060